### PR TITLE
min and max insert size

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Some atac-seq guides recommend only using fragments from the nucleosome free region (insert size smaller than ~150-200 bp). This is now implemented. The implementation is a bit mehh, since we do want to get the insert size plot, so we split the sieve result into two.

1. After all the sieving is done, except the insert size sieving. This is used for the multiqc plot insert size plot
2. After all the sieving, including insert size sieving. This is used for the remainder of the workflow

**Checklist**
- [x] I made a PR to develop (not master)
- [x] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

**Comment by Maarten-vd-Sande** [comment]
proof of concept

image

image

File: seq2science/rules/bam_cleaning.smk [hunk 1]
```diff
@@ -241,6 +262,21 @@ rule samtools_index:
         samtools index {params} {input} {output}
         """
 
+rule sam2bam:
```
Comment (siebrenf) [inline seq2science/rules/bam_cleaning.smk:265]: nothing wrong with this rule, but any error with bam files in the future will point to this rule!

File: seq2science/rules/configuration_workflows.smk [hunk 2]
```diff
@@ -62,6 +62,16 @@ if config.get("peak_caller", False):
         else:
             config["macs2_types"].extend(["summits.bed", "peaks.narrowPeak"])
 
+# make sure that both maximum and minimum insert sizes are existing when one of them is used
+if config.get("min_template_length") and not config.get("max_template_length"):
+    config["max_template_length"] = 1_000_000_000
```
Comment (siebrenf) [inline seq2science/rules/configuration_workflows.smk:67]: woah! Did not know you could write long numbers like that!

File: tests/run_tests.sh [hunk 3]
```diff
@@ -163,10 +163,10 @@ if [ $1 = "atac-seq" ]; then
 #  seq2science run atac-seq --cores $CORES --configfile tests/alignment/remote_genome_n_sample.yaml --snakemakeOptions config={aligner:bowtie2}
 
   printf "\natac-seq - multiqc report\n"
-  seq2science run atac-seq --cores $CORES --configfile tests/alignment/remote_genome_n_sample.yaml --snakemakeOptions config={aligner:bowtie2,create_qc_report:True,custom_genome_extension:tests/tinydata/tinyERCC92.fa} show_failed_logs=True
+  seq2science run atac-seq --cores $CORES --configfile tests/alignment/remote_genome_n_sample.yaml --snakemakeOptions config={aligner:bowtie2,create_qc_report:True,custom_genome_extension:tests/tinydata/tinyERCC92.fa,max_template_length:150} show_failed_logs=True
```
Comment (siebrenf) [inline tests/run_tests.sh:166]: change one of these into `min_template_length:30`, just to have a test with it too.

**Review by siebrenf — APPROVED** [review]

File: tests/run_tests.sh (hunk 3 above)
Comment (siebrenf) [reply tests/run_tests.sh]: i figured you meant to do that already and/or dont mind me, so its done

File: seq2science/rules/bam_cleaning.smk (hunk 1 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/bam_cleaning.smk:265]: Yes, maybe I will put an if statement around it

File: tests/run_tests.sh (hunk 3 above)
Comment (Maarten-vd-Sande) [reply tests/run_tests.sh]: thanks :)

**Pull request merged**